src/cache/precomputed_quantities.jl

```diff
@@ -83,6 +85,17 @@ function precomputed_quantities(Y, atmos)
             ᶜS_q_totʲs = similar(Y.c, NTuple{n, FT}),
             ᶜS_q_tot⁰ = similar(Y.c, FT),
         ) : (;)
+    sgs_quantities =
+        (
+            atmos.turbconv_model isa AbstractEDMF ||
+            isnothing(atmos.turbconv_model)
+        ) ?
```

Comment:
We don't need this check?

Reply:
Now that look at it, I think you’re right. How is that so? We always call this? With every model?

Reply:
Yes, we need this for mixing length diagnostics and cloud fraction. I guess we don't have to call it for the dry model, but if we want mixing length diagnostics we still need it.

Reply:
And right now it only happens in the callbacks?

Reply:
Yes

Reply:
Actually, to be more accurate, it only happens in the callbacks when `isnothing(atmos.turbconv_model)` is true. If `atmos.turbconv_model isa AbstractEDMF` is true, it happens at every stage.

Reply:
Ok, so we can always put it in the cache, but we can filter when it's computed (that explains the perf hit in the flame graphs)